wiki/TypeScript-Migration.md

```diff
@@ -0,0 +1,49 @@
+# TypeScript migration
+
+The following steps are recommended for migrating components to TypeScript.
+
+1. [Pick a component]((https://hashicorp.atlassian.net/browse/HDS-2392)), assign yourself to the task in Jira and add it to the current sprint
+
+2. Rename the backing class from `.js` to `.ts`; commit after this step to make it easier to review the changes
+
+3. Opening the files (`.js` and `.hbs`) in VS Code should show you the linting errors (if you have the [Glint extension](https://marketplace.visualstudio.com/items?itemName=typed-ember.glint-vscode)); similarly, if you are running `yarn lint` in `packages/components`
+
+4. Define component signature
+
+- You can either do it manually [following the documentation](https://typed-ember.gitbook.io/glint/environments/ember/component-signatures#glimmer-components)
+
+- Or, you can generate it using `npx ember-codemod-args-to-signature --src /path/to/component`. This codemod renames the component class name to `IndexComponent` by default, so make sure to revert it (e.g. `HdsBadgeIndexComponent`)
+
+5. Add types and use them in the signature
+
+- The signature should follow the naming convention (e.g. `HdsBadgeIndexSignature`) and should be placed right before the backing class declaration, so it’s consistent across components
+
+- Compare the signature generated, with the expected arguments using the website documentation (Component API section), and start defining the [basic types](https://www.typescriptlang.org/docs/handbook/basic-types.html) (e.g. `boolean`, `number`, `string`)
+
+- For more involved types, such as `enum`s, conditional and mapped types, create a `types.ts` file colocated with the component (a single file in the “root” of the component, containing types for components and sub-components)
+[See: Always place types in a separate file]
```

Comment:
the "always place types in a separate file" should link to the [Google Doc](https://docs.google.com/document/d/17T9v7pYUNrm2TtcZE2le_fbX30Jop-iG3um7b0trRlg/edit#heading=h.eyknwo6qp78t), or even better (as we agreed to unify the documentation in a single place) a section in this document

Reply:
yes, I left them in squared brackets to remind myself to port the content 🤦 
will update